marlowe-playground-server/src/Marlowe/Mermaid.hs

```diff
@@ -72,12 +96,27 @@ contractShow :: Contract -> String
 contractShow = escape . repr
   where
     repr Close        = "Close"
-    repr (When _ _ _) = "When ..."
-    repr (If o _ _ )  = observationShow o
     repr (Pay from to tok v _) =
-      "Pay from " ++ show from ++ " to " ++ show to ++ " " ++ show v ++ " of " ++ show tok
-    repr x = show x
+      "Pay " ++  valueShow v  ++ " from " ++ show from ++ " to " ++ show to
+    repr (If obs _ _ )  = observationShow obs
+    repr (When _ _ _) = "When ..."
+    repr (Let (ValueId valId) val _) = "let " ++ show valId ++ " = " ++ valueShow val
+    repr (Assert obs _) = observationShow obs
 
+valueShow :: Value -> String
+valueShow (Constant n)          = show n
+valueShow (ConstantParam param) = show param
+valueShow (AddValue val1 val2) = "(" ++ valueShow val1 ++ " + " ++ valueShow val2 ++ ")"
+valueShow (NegValue val1) = "-" ++ valueShow val1
+valueShow (SubValue val1 val2) = "(" ++ valueShow val1 ++ " - " ++ valueShow val2 ++ ")"
+valueShow (MulValue val1 val2) = "(" ++ valueShow val1 ++ " * " ++ valueShow val2 ++ ")"
+valueShow (ChoiceValue (ChoiceId id party)) = show party ++ " choice on " ++ show id
+valueShow (Cond obs val1 val2) = "(" ++ observationShow obs ++ " ? " ++ valueShow val1 ++ " : " ++ valueShow val2 ++ ")"
+valueShow (UseValue (ValueId id)) = show id
+valueShow (Scale fraction val) = "(" ++ show fraction ++ " * " ++ valueShow val ++ ")"
```

Comment:
I was thinking when I saw the diagrams you shared that maybe it would be good to display the fraction as `(fraction_numerator / fraction_denominator) * val`, instead of the default `Show` that is `fraction_numerator % fraction_denominator * val`

Reply:
done